src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/exception/StoreException.java

```diff
@@ -0,0 +1,11 @@
+package com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.exception;
+
+/**
+ * Thrown if there is a problem storing or loading values.
+ */
+public class StoreException extends RuntimeException {
```

Comment:
Is there any harm in passing in the underlying/cause exceptions here (as well as in `InvalidTokenException`)?
E.g. a `StoreException` can be caused by an `IOException`.

Reply:
There is no `ServiceProviderTokenStore` implementation. I am hoping that then we can create an additional constructor that chains the exception (I think it is mandatory, we should not eat exceptions). At the moment we are simply raising this exception with a String message.